app/src/main/java/com/devshish/internship/data/repository/LyricsRepositoryImpl.kt

```diff
@@ -1,17 +1,29 @@
 package com.devshish.internship.data.repository
 
+import com.devshish.internship.data.db.LyricsDAO
+import com.devshish.internship.data.model.room.RoomSong
 import com.devshish.internship.domain.model.SearchSong
 import com.devshish.internship.domain.repository.ILyricsRepository
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 import org.jsoup.Jsoup
 
-class LyricsRepositoryImpl : ILyricsRepository {
+class LyricsRepositoryImpl(
+    private val lyricsDAO: LyricsDAO
+) : ILyricsRepository {
 
     // TODO BUG! PAGE IS NOT ALWAYS RENDERED ENOUGH
     override suspend fun getLyrics(song: SearchSong): String =
-        withContext(Dispatchers.IO) {
-            val doc = Jsoup.connect(song.lyricsUri).get()
```

Comment:
1. You should not add lyrics to SearchSong - instead do next:
Here check if you have lyrics stored inside of database - if you have lirycs inside - return it here, else go to internet